tests/integration/test_studio_editable.py

```diff
@@ -0,0 +1,301 @@
+        block = self.load_root_xblock()
+        block.color = "green"
+        block.count = 5
+        block.comment = "Hello"
+        block.save()
+        orig_values = {field_name: getattr(block, field_name) for field_name in EditableXBlock.editable_fields}
+        # Reload the page:
+        self.go_to_view("studio_view")
+        self.fix_js_environment()
+
+        self.click_save()
+        block = self.load_root_xblock()  # Need to reload the block to bypass its cache
+        self.assert_unchanged(block, orig_values, explicitly_set=True)
+
+    def test_explicit_overrides(self):
+        """
+        Test that we can override the defaults with the same value as the default, and that the
+        value will be saved explicitly.
+        """
+        block = self.load_root_xblock()
+        self.assert_unchanged(block)
+
+        field_names = EditableXBlock.editable_fields
+        # It is crucial to this test that at least one of the fields is a String field with
+        # an empty string as its default value:
+        defaults = set([block.fields[field_name].default for field_name in field_names])
+        self.assertIn(u'', defaults)
+
+        for field_name in field_names:
+            control = self.get_element_for_field(field_name)
+            control.send_keys('9999')  # In case the field is blank and the new value is blank, this forces a change
+            control.clear()
+            control.send_keys(str(block.fields[field_name].default))
+
+        self.click_save()
+        self.assert_unchanged(block, explicitly_set=True)
+
+    def test_set_and_reset(self):
+        """
+        Test that we can set values, save, then reset to defaults.
+        """
+        block = self.load_root_xblock()
+        self.assert_unchanged(block)
+
+        for field_name in EditableXBlock.editable_fields:
+            color_control = self.get_element_for_field(field_name)
+            color_control.clear()
+            color_control.send_keys('1000')
+
+        self.click_save()
+
+        block = self.load_root_xblock()  # Need to reload the block to bypass its cache
+
+        self.assertEqual(block.color, '1000')
+        self.assertEqual(block.count, 1000)
+        self.assertEqual(block.comment, '1000')
+
+        for field_name in EditableXBlock.editable_fields:
+            self.click_reset_for_field(field_name)
+
+        self.click_save()
+        block = self.load_root_xblock()  # Need to reload the block to bypass its cache
+        self.assert_unchanged(block)
+
+    def test_invalid_data(self):
+        """
+        Test that we get notified when there's a problem with our data.
+        """
+        def expect_error_message(expected_message):
+            notification = self.dequeue_runtime_notification()
+            self.assertEqual(notification[0], "error")
+            self.assertEqual(notification[1]["title"], "Unable to update settings")
+            self.assertEqual(notification[1]["message"], expected_message)
```

Comment:
diff-cover asserts this line is uncovered, which seems quite strange if the previous two lines are covered.